<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>SettingsDialog</class>
    <widget class="QDialog" name="SettingsDialog">
        <property name="windowTitle">
            <string>Settings</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
            <!-- Visualization Options Group -->
            <item>
                <widget class="QGroupBox" name="groupBoxVisualization">
                    <property name="title">
                        <string>Visualization Options</string>
                    </property>
                    <layout class="QVBoxLayout" name="verticalLayoutVisualization">
                        <!-- Show Fourier Drawing Preview -->
                        <item>
                            <widget class="QCheckBox" name="checkBoxShowFourierPreview">
                                <property name="text">
                                    <string>Show Fourier Drawing Preview</string>
                                </property>
                            </widget>
                        </item>
                        <!-- Fourier Drawing Preview Color and Opacity -->
                        <item>
                            <layout class="QHBoxLayout" name="horizontalLayoutFourierColor">
                                <item>
                                    <widget class="QLabel" name="labelFourierColor">
                                        <property name="text">
                                            <string>Fourier Preview Color:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QPushButton" name="buttonFourierColor">
                                        <property name="text">
                                            <string>Select Color</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QLabel" name="labelFourierOpacity">
                                        <property name="text">
                                            <string>Opacity:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QSlider" name="sliderFourierOpacity">
                                        <property name="orientation">
                                            <enum>Qt::Horizontal</enum>
                                        </property>
                                        <property name="minimum">
                                            <number>0</number>
                                        </property>
                                        <property name="maximum">
                                            <number>100</number>
                                        </property>
                                        <property name="value">
                                            <number>25</number>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QLabel" name="labelFourierOpacityValue">
                                        <property name="text">
                                            <string>25%</string>
                                        </property>
                                    </widget>
                                </item>
                            </layout>
                        </item>
                        <!-- Show Drawing Tip Trail -->
                        <item>
                            <widget class="QCheckBox" name="checkBoxShowTrail">
                                <property name="text">
                                    <string>Show Drawing Tip Trail</string>
                                </property>
                            </widget>
                        </item>
                        <!-- Drawing Tip Trail Color -->
                        <item>
                            <layout class="QHBoxLayout" name="horizontalLayoutTrailColor">
                                <item>
                                    <widget class="QLabel" name="labelTrailColor">
                                        <property name="text">
                                            <string>Trail Color:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QPushButton" name="buttonTrailColor">
                                        <property name="text">
                                            <string>Select Color</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QLabel" name="labelTrailWidth">
                                        <property name="text">
                                            <string>Width:</string>
                                        </property>
                                    </widget>
                                </item>
                                <!-- Change QSpinBox to QDoubleSpinBox -->
                                <item>
                                    <widget class="QDoubleSpinBox" name="spinBoxTrailWidth">
                                        <property name="minimum">
                                            <double>0.1</double>
                                        </property>
                                        <property name="maximum">
                                            <double>10.0</double>
                                        </property>
                                        <property name="value">
                                            <double>2.0</double>
                                        </property>
                                        <property name="singleStep">
                                            <double>0.1</double>
                                        </property>
                                        <property name="decimals">
                                            <number>1</number>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QLabel" name="labelTrailWidthPx">
                                        <property name="text">
                                            <string>px</string>
                                        </property>
                                    </widget>
                                </item>
                            </layout>
                        </item>
                    </layout>
                </widget>
            </item>
            <!-- Anti-Aliasing Group -->
            <item>
                <widget class="QGroupBox" name="groupBoxAntiAliasing">
                    <property name="title">
                        <string>Anti-Aliasing</string>
                    </property>
                    <layout class="QHBoxLayout" name="horizontalLayoutAntiAliasing">
                        <item>
                            <widget class="QLabel" name="labelAntiAliasingPasses">
                                <property name="text">
                                    <string>Number of Passes:</string>
                                </property>
                            </widget>
                        </item>
                        <item>
                            <widget class="QSpinBox" name="spinBoxAntiAliasingPasses">
                                <property name="minimum">
                                    <number>1</number>
                                </property>
                                <property name="maximum">
                                    <number>10</number>
                                </property>
                                <property name="value">
                                    <number>1</number>
                                </property>
                            </widget>
                        </item>
                    </layout>
                </widget>
            </item>
            <!-- Appearance Settings Group -->
            <item>
                <widget class="QGroupBox" name="groupBoxAppearance">
                    <property name="title">
                        <string>Appearance Settings</string>
                    </property>
                    <layout class="QHBoxLayout" name="horizontalLayoutBackgroundColor">
                        <item>
                            <widget class="QLabel" name="labelBackgroundColor">
                                <property name="text">
                                    <string>Background Color:</string>
                                </property>
                            </widget>
                        </item>
                        <item>
                            <widget class="QPushButton" name="buttonBackgroundColor">
                                <property name="text">
                                    <string>Select Color</string>
                                </property>
                            </widget>
                        </item>
                    </layout>
                </widget>
            </item>
            <!-- Arrow Settings Group -->
            <item>
                <widget class="QGroupBox" name="groupBoxArrowSettings">
                    <property name="title">
                        <string>Arrow Settings</string>
                    </property>
                    <layout class="QVBoxLayout" name="verticalLayoutArrowSettings">
                        <!-- Arrow Head Base Size and Scaling Factor -->
                        <item>
                            <layout class="QHBoxLayout" name="horizontalLayoutArrowHeadSize">
                                <item>
                                    <widget class="QLabel" name="labelArrowHeadBaseSize">
                                        <property name="text">
                                            <string>Arrow Head Base Size:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QDoubleSpinBox" name="doubleSpinBoxArrowHeadBaseSize">
                                        <property name="minimum">
                                            <double>0.01</double>
                                        </property>
                                        <property name="maximum">
                                            <double>100.0</double>
                                        </property>
                                        <property name="value">
                                            <double>2.0</double>
                                        </property>
                                        <property name="singleStep">
                                            <double>0.01</double>
                                        </property>
                                        <property name="decimals">
                                            <number>2</number>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QLabel" name="labelArrowHeadScalingFactor">
                                        <property name="text">
                                            <string>Scaling Factor:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QDoubleSpinBox" name="doubleSpinBoxArrowHeadScalingFactor">
                                        <property name="minimum">
                                            <double>0.0</double>
                                        </property>
                                        <property name="maximum">
                                            <double>50.0</double>
                                        </property>
                                        <property name="value">
                                            <double>1.0</double>
                                        </property>
                                        <property name="singleStep">
                                            <double>0.1</double>
                                        </property>
                                        <property name="decimals">
                                            <number>1</number>
                                        </property>
                                    </widget>
                                </item>
                            </layout>
                        </item>
                        <!-- Arrow Line Base Width and Scaling Factor -->
                        <item>
                            <layout class="QHBoxLayout" name="horizontalLayoutArrowLineWidth">
                                <item>
                                    <widget class="QLabel" name="labelArrowLineBaseWidth">
                                        <property name="text">
                                            <string>Arrow Line Base Width:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QDoubleSpinBox" name="doubleSpinBoxArrowLineBaseWidth">
                                        <property name="minimum">
                                            <double>0.01</double>
                                        </property>
                                        <property name="maximum">
                                            <double>10.0</double>
                                        </property>
                                        <property name="value">
                                            <double>1.0</double>
                                        </property>
                                        <property name="singleStep">
                                            <double>0.01</double>
                                        </property>
                                        <property name="decimals">
                                            <number>2</number>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QLabel" name="labelArrowLineScalingFactor">
                                        <property name="text">
                                            <string>Scaling Factor:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QDoubleSpinBox" name="doubleSpinBoxArrowLineScalingFactor">
                                        <property name="minimum">
                                            <double>0.0</double>
                                        </property>
                                        <property name="maximum">
                                            <double>100.0</double>
                                        </property>
                                        <property name="value">
                                            <double>1.0</double>
                                        </property>
                                        <property name="singleStep">
                                            <double>0.1</double>
                                        </property>
                                        <property name="decimals">
                                            <number>1</number>
                                        </property>
                                    </widget>
                                </item>
                            </layout>
                        </item>
                        <!-- Arrow Maximum Line Width and Maximum Head Size -->
                        <item>
                            <layout class="QHBoxLayout" name="horizontalLayoutArrowMaxima">
                                <item>
                                    <widget class="QLabel" name="labelArrowHeadMaxSize">
                                        <property name="text">
                                            <string>Arrow Head Max Size:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QDoubleSpinBox" name="doubleSpinBoxArrowHeadMaxSize">
                                        <property name="minimum">
                                            <double>0.01</double>
                                        </property>
                                        <property name="maximum">
                                            <double>50.0</double>
                                        </property>
                                        <property name="value">
                                            <double>1.0</double>
                                        </property>
                                        <property name="singleStep">
                                            <double>0.01</double>
                                        </property>
                                        <property name="decimals">
                                            <number>2</number>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QLabel" name="labelArrowLineMaxWidth">
                                        <property name="text">
                                            <string>Arrow Line Max Width:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QDoubleSpinBox" name="doubleSpinBoxArrowLineMaxWidth">
                                        <property name="minimum">
                                            <double>0.01</double>
                                        </property>
                                        <property name="maximum">
                                            <double>20.0</double>
                                        </property>
                                        <property name="value">
                                            <double>1.0</double>
                                        </property>
                                        <property name="singleStep">
                                            <double>0.1</double>
                                        </property>
                                        <property name="decimals">
                                            <number>2</number>
                                        </property>
                                    </widget>
                                </item>
                            </layout>
                        </item>
                        <!-- Arrow Cap Type -->
                        <item>
                            <layout class="QHBoxLayout" name="horizontalLayoutArrowCapType">
                                <item>
                                    <widget class="QLabel" name="labelArrowCapType">
                                        <property name="text">
                                            <string>Arrow Cap Type:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QComboBox" name="comboBoxArrowCapType">
                                        <item>
                                            <property name="text">
                                                <string>Normal Arrow Head</string>
                                            </property>
                                        </item>
                                        <item>
                                            <property name="text">
                                                <string>No Arrow Cap</string>
                                            </property>
                                        </item>
                                        <item>
                                            <property name="text">
                                                <string>Dot</string>
                                            </property>
                                        </item>
                                    </widget>
                                </item>
                            </layout>
                        </item>
                        <!-- Arrow Color -->
                        <item>
                            <layout class="QHBoxLayout" name="horizontalLayoutArrowColor">
                                <item>
                                    <widget class="QLabel" name="labelArrowColor">
                                        <property name="text">
                                            <string>Arrow Color:</string>
                                        </property>
                                    </widget>
                                </item>
                                <item>
                                    <widget class="QPushButton" name="buttonArrowColor">
                                        <property name="text">
                                            <string>Select Color</string>
                                        </property>
                                    </widget>
                                </item>
                            </layout>
                        </item>
                    </layout>
                </widget>
            </item>
            <!-- Dialog Buttons -->
            <item>
                <layout class="QHBoxLayout" name="horizontalLayoutButtons">
                    <property name="spacing">
                        <number>20</number>
                    </property>
                    <!-- Corrected Spacer Definition -->
                    <item>
                        <spacer name="horizontalSpacer">
                            <property name="orientation">
                                <enum>Qt::Horizontal</enum>
                            </property>
                            <property name="sizeHint" stdset="0">
                                <size>
                                    <width>40</width>
                                    <height>20</height>
                                </size>
                            </property>
                        </spacer>
                    </item>
                    <!-- Save and Load Buttons -->
                    <item>
                        <widget class="QPushButton" name="buttonLoadConfig">
                            <property name="text">
                                <string>Load Config</string>
                            </property>
                        </widget>
                    </item>
                    <item>
                        <widget class="QPushButton" name="buttonSaveConfig">
                            <property name="text">
                                <string>Save Config</string>
                            </property>
                        </widget>
                    </item>
                    <!-- Dialog Buttons -->
                    <item>
                        <widget class="QPushButton" name="buttonCancel">
                            <property name="text">
                                <string>Cancel</string>
                            </property>
                        </widget>
                    </item>
                    <item>
                        <widget class="QPushButton" name="buttonApply">
                            <property name="text">
                                <string>Apply</string>
                            </property>
                        </widget>
                    </item>
                    <item>
                        <widget class="QPushButton" name="buttonOK">
                            <property name="text">
                                <string>OK</string>
                            </property>
                        </widget>
                    </item>
                </layout>
            </item>
        </layout>
    </widget>
    <resources/>
    <connections/>
</ui>